modules/plot/main/coffee/Axis.coffee

```diff
@@ -321,60 +318,20 @@ class Axis
       @yScale.domain(domain)
     else
 
-      # OPTIM: this expensive calculation is not needed if the scales are non auto
-      # XXX: band series?
-
-      ymin = undefined
-      ymax = undefined
-
-      types = hx.groupBy(@series(), (d) -> d._.type)
-      stackGroups = types.map (d) ->
-        type: d[0]
-        group: hx.groupBy(d[1], (s) -> if supportsGroup(s) then s.group() else undefined)
-
-      for type in stackGroups
-        for group in type.group
-          series = group[1]
-          if group[0] == undefined
-
-            ys = for s in @series()
-              data = s.data()
-              if s instanceof StraightLineSeries
-                if not data.dx and not data.dy and data.y
-                  [data.y, data.y]
-                else
-                  undefined
-              else if s instanceof BandSeries
-                extent2(data, ((d) -> d.y1), (d) -> d.y2)
-              else
-                extent(data, (d) -> d.y)
-            ys = ys.filter((d) -> d?)
+      { yMinMightBeAuto, yMaxMightBeAuto } = { yMaxMightBeAuto: @y.max(), yMinMightBeAuto: @y.min() }
+      { ymin, ymax } = if 'auto' in [yMaxMightBeAuto, yMinMightBeAuto]
+        { ymin, ymax } = @calculateYBounds()
+        yminscaled = if yMinMightBeAuto is 'auto' then scalePad(ymin, ymax - ymin, -@y.scalePaddingMin()) else yMinMightBeAuto
+        ymaxscaled = if yMaxMightBeAuto is 'auto' then scalePad(ymax, ymax - ymin, @y.scalePaddingMax()) else yMaxMightBeAuto
+        { ymin: yminscaled, ymax: ymaxscaled }
+      else
+        { ymin: yMinMightBeAuto, ymax: yMaxMightBeAuto }
 
-            yymin = hx.min(ys.map((d) -> d[0]))
-            yymax = hx.max(ys.map((d) -> d[1]))
-            if (ymin == undefined or yymin < ymin) then ymin = yymin
-            if (ymax == undefined or yymax > ymax) then ymax = yymax
-          else
-            topSeries = series[series.length-1]
-            if ymin == undefined
-              ymin = 0
-            else
-              ymin = Math.min(ymin, 0)
-            if ymax == undefined
-              ymax = 0
-            else
-              ymax = Math.max(ymax, 0)
-            for d in topSeries.data()
-              stackHeight = @getYStack(topSeries._.type, topSeries.group(), d.x, topSeries._.seriesId+1, @yScale.domainMin)
-              if (ymin == undefined or stackHeight < ymin) then ymin = stackHeight
-              if (ymax == undefined or stackHeight > ymax) then ymax = stackHeight
 
       ymin = if @y.min() is 'auto' then ymin else @y.min()
       ymax = if @y.max() is 'auto' then ymax else @y.max()
```

Comment:
You may well be right, I'd like to make a test just to make sure though.

I think the easiest way to tell a non-auto value is not ignored is by moving this logic into calculateYBounds and writing a test for it there.